### ISSUE TITLE
`./wpt run --verify` claims flakiness where there are duplicate test names

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt#14645]
https://wpt.fyi/results/css/geometry/interfaces.worker.html?label=stable&aligned shows that the test has a duplicate test name "Test default toJSON operation of toJSON object" in Edge and Firefox. That it is implementation-dependent is surprising and wrong, but all that matters for this issue is that there are duplicate test names.

https://taskcluster-artifacts.net/BX_K1_06R7OU-LiJjKyYmQ/0/public/logs/live_backing.log for https://github.com/web-platform-tests/wpt/pull/14629 claimed that the test was flaky:

|                    Test                   |                                                     Subtest                                                      |            Results            |                                                              Messages                                                              |
|-------------------------------------------|------------------------------------------------------------------------------------------------------------------|-------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| `/css/geometry/interfaces.worker.html`    | `Test default toJSON operation of toJSON object`                                                                 | **FAIL: 60/10**               | `memberHolderObject is undefined`                                                                                                  |

However, FAIL is the only result, and as indicated by "60/10" there's probably a bug in the `./wpt run --verify` code here, where the tests are overcounted and anything other than "10/10" is treated as flakiness.

**Comment by jgraham** [comment]
Having duplicate test names isn't allowed, so in the absence of anything else catching this, the flakiness check is the best place (it's a form of flakiness because infrastructure will confuse the results of multiple tests, so even if it's not flaky now it's likely to become so in the future and wrong in any case).

**Comment by foolip** [comment]
Alright, let's make this issue about two things:
- some else blocking PRs on harness errors, probably after first taking an inventory of harness errors
- requiring more than one unique result to consider something flaky in `./wpt run --verify`

Letting the first block the second seems fine.

**Comment by foolip** [comment]
https://github.com/web-platform-tests/wpt/pull/14841 was affected by this.

**Comment by foolip** [comment]
@Ms2ger I guess https://github.com/web-platform-tests/wpt/pull/14866 resolved this in the only case we knew it happened, or do you know of others?

**Comment by Ms2ger** [comment]
I think this issue isn't about fixing the existing duplicated test names, per https://github.com/web-platform-tests/wpt/issues/14645#issuecomment-449445728?

**Comment by foolip** [comment]
@tabatkins ran into this as well and asked me about it.

@jgraham if we write some code that explicitly looks for duplicate test names from a single run, do you think it makes sense together with `--verify`? Another alternative would be to do it in the wpt.fyi checks. @lukebjerring WDYT?

**Comment by lukebjerring** [comment]
wpt.fyi discovers these things too late, IMHO. Because of problems we have with noise of the checks, we'd end up ignoring it. Special-casing duplication is a better approach.

**Comment by tabatkins** [comment]
Yeah, when I ran into this it confused the heck out of me. I thought the `20/10` I saw was a display bug and meant to say `2/10`, and spent a while very confused about how that particular test could possibly have flaked.

There *is* an error message about the duplicate test name in the output, but it's so buried among the massive amount of other logging that I didn't discover it until after I'd already figured out the problem.

This *really* needs to be special cased in the output of the table, as it fundamentally is *not* a flakiness issue, but something totally different.
